Android/app/src/main/java/com/example/instagramclone/Activities/EditProfileActivity.java

```diff
@@ -0,0 +1,186 @@
+package com.example.instagramclone.Activities;
+
+import androidx.annotation.Nullable;
+import androidx.appcompat.app.AppCompatActivity;
+
+import android.content.Intent;
+import android.graphics.Bitmap;
+import android.graphics.BitmapFactory;
+import android.net.Uri;
+import android.os.Bundle;
+import android.preference.PreferenceManager;
+import android.provider.MediaStore;
+import android.util.Log;
+import android.view.View;
+import android.widget.ImageView;
+import android.widget.TextView;
+
+import com.example.instagramclone.Api.UserApi;
+import com.example.instagramclone.Common.APIClient;
+import com.example.instagramclone.Common.FileUtil;
+import com.example.instagramclone.Models.CurrentUserModel;
+import com.example.instagramclone.Models.FeedItemModel;
+import com.example.instagramclone.Models.UserModel;
+import com.example.instagramclone.Models.UserObjModel;
+import com.example.instagramclone.R;
+
+import java.io.File;
+
+import okhttp3.MediaType;
+import okhttp3.MultipartBody;
+import okhttp3.RequestBody;
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
+
+public class EditProfileActivity extends AppCompatActivity {
+    private static final String TAG = "EditProfileActivity";
+    private static final int PICK_IMAGE = 100;
+
+    private ImageView cancel, done, avatar;
+    private TextView editAvatarText, fullname, username, description;
+
+    private UserApi userApi;
+
+    private boolean newAvatar = false;
+
+    private String avatarUri;
+
+    private UserModel user;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_edit_profile);
+        initComponents();
+    }
+
+    private void initComponents() {
+        cancel = findViewById(R.id.edit_profile_toolbar_cancel);
+        done = findViewById(R.id.edit_profile_toolbar_done);
+        avatar = findViewById(R.id.edit_profile_avatar);
+        editAvatarText = findViewById(R.id.edit_avatar_text);
+        fullname = findViewById(R.id.edit_text_name);
+        username = findViewById(R.id.edit_text_username);
+        description = findViewById(R.id.edit_text_description);
+        userApi = APIClient.getClient().create(UserApi.class);
+
+        initData();
+    }
+
+    private void initData() {
+        user = CurrentUserModel.getInstance();
```

Comment:
`CurrentUserModel` ? If it is a singleton that holds user data it should be just CurrentUser